server/images.go

```diff
@@ -1187,7 +1192,10 @@ func makeRequestWithRetry(ctx context.Context, method string, requestURL *url.UR
 				body.Seek(0, io.SeekStart)
 			}
 
-			continue
+			if try == (maxRetries - 1) {
```

Comment:
could we set an outer `err` variable instead and then just return it at the end? feels strange to check boundary conditions

Reply:
I agree. It's cleaner to set a `lastError` and return. Other similar functions follow that pattern already.

```go
var lastError error
for try := 0; try < maxRetries; try++ {
  ...
  switch resp.StatusCode {
  case http.StatusUnauthorized:
    lastError = fmt.Errorf("unauthorized: make sure you have access to this model")
    continue
  }
}

return nil, fmt.Errorf("%w: %w", errMaxRetriesExceeded, lastError)
```

This also enables the function to handle retries for other status codes

Reply:
Made this change 👍